datafusion/core/src/datasource/listing/table.rs

```diff
@@ -67,6 +67,10 @@ pub struct ListingTableConfig {
     pub file_schema: Option<SchemaRef>,
     /// Optional `ListingOptions` for the to be created `ListingTable`.
     pub options: Option<ListingOptions>,
+    /// Optional default is false, if temporary is true, then it means that
+    /// we will create a temporary table for select * from `xxx.file`
+    /// or describe 'xxx.file', the temporary table will be registered to the schema
+    pub temporary_file: bool,
```

Comment:
💯  for the comments. Thank you @xiaoyong-z 

I think the term `temporary_files` often means something different -- specifically files in `/tmp` or similar -- so it may be confusing to use the same term. 

What would you think about using a different term than `temporary`? Perhaps `ephemeral` would be appropriate.

Reply:
update a pr, now it's no need to use temporary_files any more